File: src/models/resource.rs
use uuid::Uuid;

use schema::resources;


#[derive(Debug, Serialize, Deserialize, Queryable)]
pub struct Resource {
    pub id: Uuid,
    pub name: String,
    pub max_amount: i64,
    pub pod_id: Uuid,
    pub base_id: Uuid,
}


#[derive(Insertable)]
#[table_name="resources"]
pub struct NewResource {
    pub name: String,
    pub max_amount: i64,
    pub pod_id: Uuid,
    pub base_id: Uuid,
}

File: server/schemas/resource.py
"""Resource schema."""
from marshmallow import fields
from server.schemas import Schema


class ResourceSchema(Schema):
    """Resource schema."""

    id = fields.UUID()
    pod_id = fields.UUID()
    base_id = fields.UUID()

    name = fields.Str(required=True)
    amount = fields.Int()
    production = fields.Int()
    max_amount = fields.Int()
    empty_at = fields.DateTime()
    last_update = fields.DateTime()
